### PR TITLE
Fix Aggregation Iteration

### PR DESCRIPTION
**Pull request by b-butler**
## Description
Don't iterate over the same aggregate twice no matter how many times it appears in different `AggregateStores`.

## Motivation and Context
Resolves: #688.

## Checklist:

- [x] I am familiar with the [Contributing Guidelines](https://github.com/glotzerlab/signac-flow/blob/master/CONTRIBUTING.md).
- [x] I agree with the terms of the [Contributor Agreement](https://github.com/glotzerlab/signac-flow/blob/master/ContributorAgreement.md).
- [x] My name is on the [list of contributors](https://github.com/glotzerlab/signac-flow/blob/master/contributors.yaml).
- [x] The changes introduced by this pull request are covered by existing or newly introduced tests.
- [x] The [package documentation](https://github.com/glotzerlab/signac-flow/tree/master/doc) and [framework documentation](https://docs.signac.io/) in [signac-docs](https://github.com/glotzerlab/signac-docs) are up to date with these changes.
- [x] I have updated the [changelog](https://github.com/glotzerlab/signac-flow/blob/master/changelog.txt) and added any related issue and pull request numbers for future reference.

### CODE REVIEW

**Comment by tommy-waltmann** [comment]
How has this been tested? The changes look good I just want to confirm that this fixes the problems from the examples in #688

**Comment by tommy-waltmann** [comment]
Let me know if/when this is ready for another round of review

**Comment by b-butler** [comment]
@tommy-waltmann it is ready.

File: tests/test_project.py [hunk 1]
```diff
@@ -2056,41 +2056,6 @@ def setup_main_interface(self, request):
...
-        # All operations will return aggregates, even if the aggregates are not
-        # unique to that operation, because every operation/group in this
-        # project has a custom aggregator defined. Only the default aggregator
-        # can de-duplicate the returned results in the cursor, thus it is
-        # expected that the length of the cursor is larger than the number of
-        # unique ids present in the cursor.
-        assert len(agg_cursor) == 42
-        assert len({get_aggregate_id(agg) for agg in agg_cursor}) == 34
-        assert tuple(project) in agg_cursor
-        assert all((job,) in agg_cursor for job in project)
```
Comment (tommy-waltmann) [inline tests/test_project.py:2073]: Has all the behavior being tested in this test been replicated in the new test class?

**Review by tommy-waltmann — COMMENTED** [review]
Just one quick question

File: flow/aggregates.py [hunk 2]
```diff
@@ -654,11 +654,18 @@ def __contains__(self, aggregate):
...
 
     def __iter__(self):
+        existing_stores = set()
         for aggregate_store in self._stores:
-            yield from aggregate_store.values()
+            for agg_id, aggregate in aggregate_store.items():
+                if agg_id in existing_stores:
+                    continue
+                existing_stores.add(agg_id)
+                yield aggregate
```
Comment (bdice) [inline flow/aggregates.py:668]:
```suggestion
                if agg_id not in existing_stores:
                    existing_stores.add(agg_id)
                    yield aggregate
```

**Review by bdice — APPROVED** [review]
Thanks for the fix. This looks fine to me.

File: tests/test_project.py (hunk 1 above)
Comment (bdice) [reply tests/test_project.py:2073]: Yes, I think @b-butler tested the core functionality correctly. I pushed a commit to further expand coverage: 17ff07f9a8a917eb19b5c957eb9129e1073c1007

**Review by bdice — APPROVED** [review]

**Pull request merged**